lib/shared/widgets/menus/character_context_menu/grind_optimizer.widget.dart

```diff
@@ -37,37 +38,54 @@ class CharacterGrindOptimizerWidget extends StatelessWidget {
     if (classType == null) return Container();
     final state = context.watch<ContextMenuOptionsBloc>();
     final currentAverage = state.getCurrentAverage(classType);
-    final achievableAverage = state.getAchievableAverage(classType);
+    if (currentAverage == null) return Container();
+    final achievableAverage = state.getAchievableAverage(classType) ?? 0;
+    final achievableDiff = achievableAverage.floor() - currentAverage.floor();
     final isInPinnacle = state.achievedPinnacleTier(classType);
     final goForReward = state.goForReward(classType);
-    final message =
+    final gofor_message =
```

Comment:
```suggestion
    final goForMessage =
```
dart conventions

Reply:
My bad! I knew you (and Dart) were using camelCase and I needed to keep it consistent, but I'm used to Python and forgot